### ISSUE TITLE
[FR] Packer free expanding in one dimension

### ISSUE THREAD

**Issue by JohannesHoffmann** [soimy/maxrects-packer#51]
I tried different width and heights of a packer like 1500 or 800. But each bins width and height is reduced to the closest number of bineric like 800 is down to 512 and 1500 down to 1024.

Maybe it is not the purpose of this library but i need an option to force the algorithm to use the complete width or the complete height. Is this possible?

Enhancement despide if yes or no: describe this behavior in the readme.

Thanks!

**Comment by JohannesHoffmann** [comment]
After a bit of research i understand it more clearly. The main use for this library is to pack rects into a bigger square rect to use this as a textrue or something.

My usecase is very different. I need material utilization. Have a roll of paper with a width of 1500mm. Whats the minimum length i need to set X rects, each rect in different size. I need to fill the complete width and the height results on how many rects can be packed.

Maybe i am the only one needing this behavior?

**Comment by soimy** [comment]
This is because the default packing options are: 
- `options.smart` packing with smallest possible size. (default is true)
- `options.pot` bin size round up to smallest power of 2. (default is true)

For your user case, only fix one dimension of a packer is not implemented, but doable.

**Comment by JohannesHoffmann** [comment]
Do you accept PRs? If yes i would implement this feature. Ideas of the option naming?

**Comment by soimy** [comment]
Of course, Feel free to submit PRs. we can discuss more in the commits

**Comment by zardoy** [comment]
@soimy the change seems unreleased. Can you release it, please? I would like to test.

@JohannesHoffmann Also I wonder why does it work only when square and pot are `false`? In my case I'm using it for rendering and I have limits on texture index in the shader code (which expects it to go from left to right, not from top to bottom). In other words, default `pot` and `square` behavior is what I basically need, but I need it to fill the textures horizontally, not vertically (textures are the same size).

**Comment by soimy** [comment]
Close via ff875a5db2896bbdf8819aef730b4ea9d10fd759
